indexing-service/src/test/java/org/apache/druid/indexing/common/task/batch/parallel/HashPartitionTaskKillTest.java

```diff
@@ -0,0 +1,435 @@
+{
+  private static final TimestampSpec TIMESTAMP_SPEC = new TimestampSpec("ts", "auto", null);
+  private static final DimensionsSpec DIMENSIONS_SPEC = new DimensionsSpec(
+      DimensionsSpec.getDefaultSchemas(Arrays.asList("ts", "dim1", "dim2"))
+  );
+  private static final InputFormat INPUT_FORMAT = new CsvInputFormat(
+      Arrays.asList("ts", "dim1", "dim2", "val"),
+      null,
+      false,
+      false,
+      0
+  );
+  private static final Interval INTERVAL_TO_INDEX = Intervals.of("2017-12/P1M");
+
+  private File inputDir;
+
+
+  public HashPartitionTaskKillTest()
+  {
+    super(LockGranularity.TIME_CHUNK, true, 0, 0);
+  }
+
+  @Before
+  public void setup() throws IOException
+  {
+    inputDir = temporaryFolder.newFolder("data");
+    final Set<Interval> intervals = new HashSet<>();
+    // set up data
+    for (int i = 0; i < 10; i++) {
+      try (final Writer writer =
+               Files.newBufferedWriter(new File(inputDir, "test_" + i).toPath(), StandardCharsets.UTF_8)) {
+        for (int j = 0; j < 10; j++) {
+          writer.write(StringUtils.format("2017-12-%d,%d,%d th test file\n", j + 1, i + 10, i));
+          writer.write(StringUtils.format("2017-12-%d,%d,%d th test file\n", j + 2, i + 11, i));
+          intervals.add(SEGMENT_GRANULARITY.bucket(DateTimes.of(StringUtils.format("2017-12-%d", j + 1))));
+          intervals.add(SEGMENT_GRANULARITY.bucket(DateTimes.of(StringUtils.format("2017-12-%d", j + 2))));
+        }
+      }
+    }
+
+    for (int i = 0; i < 5; i++) {
+      try (final Writer writer =
+               Files.newBufferedWriter(new File(inputDir, "filtered_" + i).toPath(), StandardCharsets.UTF_8)) {
+        writer.write(StringUtils.format("2017-12-%d,%d,%d th test file\n", i + 1, i + 10, i));
+      }
+    }
+    // sorted input intervals
+    List<Interval> inputIntervals = new ArrayList<>(intervals);
+    inputIntervals.sort(Comparators.intervalsByStartThenEnd());
+  }
+
+  @Test(timeout = 5000L)
+  public void failsInFirstPhase() throws Exception
+  {
+    final ParallelIndexSupervisorTask task =
+        newTask(TIMESTAMP_SPEC, DIMENSIONS_SPEC, INPUT_FORMAT, null, INTERVAL_TO_INDEX, inputDir,
+                "test_*",
+                new HashedPartitionsSpec(null, null, // num shards is null to force it to go to first phase
+                                         ImmutableList.of("dim1", "dim2")
+                ),
+                2, false, true, 0
+        );
+
+    final TaskActionClient actionClient = createActionClient(task);
+    final TaskToolbox toolbox = createTaskToolbox(task, actionClient);
+
+    prepareTaskForLocking(task);
+    Assert.assertTrue(task.isReady(actionClient));
+    task.stopGracefully(null);
+
+    TaskStatus taskStatus = task.runHashPartitionMultiPhaseParallel(toolbox);
+
+    Assert.assertTrue(taskStatus.isFailure());
+    Assert.assertEquals(
+        "Failed in phase[TestRunner[false]]. See task logs for details.",
```

Comment:
I'm unsure what the failed phase was because its name is always same. I suggest to name each phase differently, so that these tests can verify the exact message.

Reply:
Phases are more evident now after my last commit